sklearn/utils/_random.pyx

```diff
@@ -278,7 +275,7 @@ cpdef sample_without_replacement(np.int_t n_population,
 
     all_methods = ("auto", "tracking_selection", "reservoir_sampling", "pool")
 
-    ratio = n_samples / n_population if n_population != 0.0 else 1.0
+    ratio = <double> n_samples / n_population if n_population != 0.0 else 1.0
```

Comment:
this function does a division between 2 integers. Setting cdivision=True would make it an integer division which is not what we want, so we use explicit casting.

I checked each division in each file where we didn't have the cdivision=True directive to make sure that there's no other case like this one.